src/types.ts

```diff
@@ -0,0 +1,153 @@
+import { TRAM_TAG, TRAM_TAG_REACTION, TRAM_TAG_NEW_EFFECTS, TRAM_TAG_CLEANUP_EFFECTS } from './node-names';
+
+/* ============= PUBLIC TYPES ========================================
+ * A lot of the types here are wrapped using an array / index of 0.
+ * This allows us to expose the types as readable on the consumption side.
+ * =====================================================================
+ */
+
+/**
+ * Type for when we can take a CSS Selector, or an HTML Element (mostly mounting).
+ */
+export type ElementOrSelector = [string | HTMLElement][0];
```

Comment:
I'm not 100% sure about this pattern, but it allows consumers to understand what the library is looking for (rather than just saying "ElementOrSelector")